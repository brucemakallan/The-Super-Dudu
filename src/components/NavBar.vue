<template>
  <nav>
    <div>NavBar</div>
  </nav>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  name: 'NavBar'
})
</script>

<style lang="scss" scoped>

</style>

