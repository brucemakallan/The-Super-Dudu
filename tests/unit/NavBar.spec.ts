import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import NavBar from '@/components/NavBar.vue';
import { mockState, mockGetters, mockMutations, mockActions } from '../mocks';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({
  state: mockState,
  getters: mockGetters,
  mutations: mockMutations,
  actions: mockActions,
});

let wrapper: any;

describe('NavBar.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(NavBar, {
      store,
      localVue,
    });
  });

  it('renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
