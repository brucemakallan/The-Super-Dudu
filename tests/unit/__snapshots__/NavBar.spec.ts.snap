// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue renders props.msg when passed 1`] = `
<nav>
  <div>NavBar</div>
</nav>
`;
